### PR TITLE
Convert StaggeredInfiniteScrollBrickFragment to kotlin

### LINKED ISSUE #211: StaggeredInfiniteScrollBrickFragment should be in kotlin

### PR DESCRIPTION
**Pull request by thusson13**
Fixes https://github.com/wayfair/brickkit-android/issues/211

### CODE REVIEW

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/staggeredinfinitescroll/StaggeredInfiniteScrollBrickFragment.kt [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        super.onPause()
+
+        dataManager.setOnReachedItemAtPosition(null)
+    }
+
+    private fun addNewBricks() {
+        dataManager.addLast(
+            (0 until PAGE_SIZE).map { i ->
+                TextBrick(
+                    if (i % 19 == 0) FullWidthBrickSize() else HalfWidthBrickSize(),
```
Comment (kenyee) [inline BrickKit/app/src/main/java/com/wayfair/brickkitdemo/staggeredinfinitescroll/StaggeredInfiniteScrollBrickFragment.kt]: 19 and 4 should be in constants?

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/staggeredinfinitescroll/StaggeredInfiniteScrollBrickFragment.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        addNewBricks()
+    }
+
+    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? =
+        super.onCreateView(inflater, container, savedInstanceState)?.apply {
+            findViewById<RecyclerView>(R.id.recycler_view).layoutManager =
+                StaggeredGridLayoutManager(2, StaggeredGridLayoutManager.VERTICAL).apply {
```
Comment (kenyee) [inline BrickKit/app/src/main/java/com/wayfair/brickkitdemo/staggeredinfinitescroll/StaggeredInfiniteScrollBrickFragment.kt]: 2 should be in a constant?

**Review by kenyee — APPROVED** [review]

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/staggeredinfinitescroll/StaggeredInfiniteScrollBrickFragment.kt (hunk 2 above)
Comment (thusson13) [reply BrickKit/app/src/main/java/com/wayfair/brickkitdemo/staggeredinfinitescroll/StaggeredInfiniteScrollBrickFragment.kt]: done

File: BrickKit/app/src/main/java/com/wayfair/brickkitdemo/staggeredinfinitescroll/StaggeredInfiniteScrollBrickFragment.kt (hunk 1 above)
Comment (thusson13) [reply BrickKit/app/src/main/java/com/wayfair/brickkitdemo/staggeredinfinitescroll/StaggeredInfiniteScrollBrickFragment.kt]: done

**Pull request merged**